cmd/lncli/cmd_pay.go

```diff
@@ -336,7 +343,13 @@ func sendPaymentRequest(ctx *cli.Context,
 	}
 
 	req.CltvLimit = int32(ctx.Int(cltvLimitFlag.Name))
-	req.TimeoutSeconds = paymentTimeoutSeconds
+
+	pmtTimeout := ctx.Duration("timeout")
+	if pmtTimeout == 0 {
```

Comment:
should check not negative?

Reply:
indeed, didn't think that `-1s` was possible input for a duration flag, but it is, check added